compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/RegExpLexer.java

```diff
@@ -241,13 +254,8 @@ private STToken readTokenInReEscape() {
         boolean isReSyntaxChar = isReSyntaxChar(nextToken);
         boolean isReSimpleCharClassCode = isReSimpleCharClassCode(nextToken);
 
-        if (!isReSyntaxChar && !isReSimpleCharClassCode) {
-            reportLexerError(DiagnosticErrorCode.ERROR_INVALID_ESCAPE_SEQUENCE);
-        }
-
         this.reader.advance();
         endMode();
-        switchParserModeFollowingReEscape();
 
         if (isReSyntaxChar) {
             return getRegExpText(SyntaxKind.RE_SYNTAX_CHAR);
```

Comment:
I think we can remove this lexer mode(readTokenInReEscape) as well. Instead of using `RE_SYNTAX_CHAR` kind we can let it be tokenized in the `RE_DISJUNCTION` mode. So we will have `QUESTION_MARK_TOKEN`, `PIP_TOKEN`, etc.

Reply:
With using `RE_DISJUNCTION` for this part `ReSimpleCharClassCode` will be tokenized as a `RE_LITERAL_CHAR`. In parser we can check the literal value and create a `RE_SIMPLE_CHAR_CLASS_CODE` token. We do something similar for the `key` keyword in the ballerina lexer. In the lexer, we tokenize it as a identifier and then in the parser we convert it to a keyword.  Ref: https://github.com/ballerina-platform/ballerina-lang/blob/87b36c4b75e86928ecd28a4254df5f4cbd200d1a/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java#L10924-L10927